module/VuFind/src/VuFind/View/Helper/Root/AccountMenu.php

```diff
@@ -0,0 +1,377 @@
+     */
+    public function checkHolds(): bool
+    {
+        return $this->checkIlsCapability('getMyHolds');
+    }
+
+    /**
+     * Check whether to show storageRetrievalRequests item
+     *
+     * @return bool
+     */
+    public function checkStorageRetrievalRequests(): bool
+    {
+        return $this->checkIlsFunction('StorageRetrievalRequests');
+    }
+
+    /**
+     * Check whether to show ILLRequests item
+     *
+     * @return bool
+     */
+    public function checkILLRequests(): bool
+    {
+        return $this->checkIlsFunction('ILLRequests');
+    }
+
+    /**
+     * Check whether to show fines item
+     *
+     * @return bool
+     */
+    public function checkFines(): bool
+    {
+        return $this->checkIlsCapability('getMyFines');
+    }
+
+    /**
+     * Check whether to show librarycards item
+     *
+     * @return bool
+     */
+    public function checkLibraryCards(): bool
+    {
+        $user = $this->getAuthHelper()->isLoggedIn();
+        return $this->isIlsOnline() && $user && $user->libraryCardsEnabled();
+    }
+
+    /**
+     * Check whether to show overdrive item
+     *
+     * @return bool
+     */
+    public function checkOverdrive(): bool
+    {
+        return $this->getView()->plugin('overdrive')->showMyContentLink();
+    }
+
+    /**
+     * Check whether to show searchhistory item
+     *
+     * @return bool
+     */
+    public function checkHistory(): bool
+    {
+        return $this->getView()->plugin('accountCapabilities')()->getSavedSearchSetting() === 'enabled';
+    }
+
+    /**
+     * Check whether to show logout item
+     *
+     * @return bool
+     */
+    public function checkLogout(): bool
+    {
+        return (bool)$this->getAuthHelper()->isLoggedIn();
+    }
+
+    /**
+     * Check ILS connection capability
+     *
+     * @param string $capability Name of then ILS method to check
+     *
+     * @return bool
+     */
+    protected function checkIlsCapability(string $capability): bool
+    {
+        return $this->isIlsOnline()
+            && $this->getIlsConnection()->checkCapability($capability, $this->getCapabilityParams());
+    }
+
+    /**
+     * Check ILS function capability
+     *
+     * @param string $function The name of the ILS function to check.
+     *
+     * @return bool
+     */
+    protected function checkIlsFunction(string $function): bool
+    {
+        return $this->isIlsOnline()
+            && $this->getIlsConnection()->checkFunction($function, $this->getCapabilityParams());
+    }
+
+    /**
+     * Check whether the ILS connection is available
+     *
+     * @return bool
+     */
+    protected function isIlsOnline(): bool
+    {
+        return 'ils-none' !== $this->getIlsConnection()->getOfflineMode();
+    }
+
+    /**
+     * Get params for checking ILS capability/function
+     *
+     * @return array
+     */
+    protected function getCapabilityParams(): array
+    {
+        $user = $this->getAuthHelper()->isLoggedIn();
```

Comment:
Not strictly necessary, but maybe it's worth adding a getUser support method to wrap around the isLoggedIn calls for greater clarity.

Reply:
done